File: experiments/source/ocaml/mixed_biased_l2_norm.ml
open Loom

let[@loom.entry] mixed_biased_l2_norm (scale : float) (bias : float)
    (x : Tensor1.t) =
  Tensor1.reduce_sum
    (Tensor1.map
       (fun xi ->
         let value = (scale *. xi) +. bias in
         value *. value)
       x)

File: experiments/source/python/mixed_biased_l2_norm.py
import loom


@loom.entry
def mixed_biased_l2_norm(scale: float, bias: float, x: loom.Tensor1) -> float:
    def square_shifted(xi: float) -> float:
        value = (scale * xi) + bias
        return value * value

    return loom.Tensor1.reduce_sum(loom.Tensor1.map(square_shifted, x))
